src/Transport/Seam/MessagePump.cs

```diff
@@ -22,10 +23,11 @@ class MessagePump : IPushMessages, IDisposable
         SatelliteTransportAddressCollection satelliteTransportAddresses;
         SemaphoreSlim throttler;
 
-        public MessagePump(ITopologySectionManager topologySectionManager, ITransportPartsContainer container, ReadOnlySettings settings)
+        public MessagePump(ITopologySectionManager topologySectionManager, ITransportPartsContainer container, INamespacePartitioningStrategy namespacePartitioningStrategy, ReadOnlySettings settings)
```

Comment:
Why is this INamespacePartitioningStrategy instead of IPerformNamespacePartitioningFailOver